feat: Update API model, BottomNavigation, and database structure

victor-chuprakov opened the pull request:
- Updated the API model and Mapper for better data handling.
- Revamped the BottomNavigation component for improved UI/UX.
- Split the database into two entities and repositories for better organization.
- Implemented a new architecture for the database.

Commit:
feat: Update API model, BottomNavigation, and database structure

- Updated the API model and Mapper for better data handling.
- Revamped the BottomNavigation component for improved UI/UX.
- Split the database into two entities and repositories for better organization.
- Implemented a new architecture for the database.

AlexKrayVR commented on app/src/main/java/com/example/foodie/common/data/mapper/Mapper.kt:
```diff
@@ -5,86 +5,62 @@ import com.example.foodie.common.data.model.FAT
 
-fun RecipeDTO.toRecipe(): Recipe {
+fun RecipeDTO?.toRecipe(): Recipe {
+    if (this == null) {
```
rework creation depending of null parameters and not of null class
example: this?.X ?: Y
it will reduce the amount of boilerplate code

AlexKrayVR commented on app/src/main/java/com/example/foodie/common/data/modelDTO/LARGEDTO.kt:
```diff
@@ -1,4 +1,4 @@
-package com.example.foodie.common.data.model
+package com.example.foodie.common.data.modelDTO
 
 data class LARGEDTO(
```
naming: LargeDTO

AlexKrayVR commented on app/src/main/java/com/example/foodie/common/data/repository/FavoriteRecipesRepositoryImpl.kt:
```diff
@@ -0,0 +1,24 @@
+        favoriteRecipesDao.deleteByUri(uri)
+    }
+
+    override suspend fun findRecipeById(id: String): FavoriteRecipe? {
```
try use kotlin style "type inference"
example:
fun x():y {return d}  -> fun x() = d

AlexKrayVR commented on app/src/main/java/com/example/foodie/common/ui/SharedViewModel.kt:
```diff
@@ -148,21 +151,18 @@ class SharedViewModel @Inject constructor(
+            launch { dataPreference.saveMealType(mealType) }
 
             // Обновляем значения после завершения операций
             _lastTime.value = time
```
update values will occur before the execution of coroutine above
use job to wait 
job = launch{}
job.join()

AlexKrayVR commented on app/src/main/java/com/example/foodie/dishes/ui/components/FoodDishCard.kt:
```diff
@@ -51,15 +51,13 @@ import com.example.foodie.common.ui.components.dishes_card_components.RecipeLabe
     val isFavorite = favoriteRecipes.any { it.uri == hit.recipe.uri }
     val newFavoriteState = remember { mutableStateOf(isFavorite) }
+
     LaunchedEffect(favoriteRecipesState.value) {
```
use remember with key to hold value that could be changed

AlexKrayVR reviewed (commented):
need refactor

Commit:
Update and simplify code:
- Made mappers for DTOs easier to read by using null checks and default values.
- Applied Kotlin's type inference in functions for better code readability.
- Used Job objects to improve filter saving in the ViewModel.
- Implemented remember with keys to track if a recipe is a favorite, so the UI updates automatically.
